### PR TITLE
ENH: Improve coverage for Strain module classes.

### PR DESCRIPTION
**Pull request by jhlegarreta**
Exercise basic object methods.

Test all Set/Get methods through the TEST_SET_GET_VALUE macro.

Use the TRY_EXPECT_NO_EXCEPTION macro for filter Update() calls to save
typing try/catch blocks.

Exercise exceptions using the TRY_EXPECT_EXCEPTION macro.

Remove duplicate call to the SetStrainFrom method in
itkTransformToStrainFilterTest.cxx.

Remove the GetDirection() and immediate SetDirection() calls for the
transformToStrainFilterObject in itkTransformToStrainFilterTest.cxx. It
does not make sense to get a property from an object to immediately set
the same property object to it.

Remove unnecessary calls to Print() in
itkTransformToStrainFilterTest.cxx (e.g. printing the
TransformStrainFilter's Transform property). Also, the macro
EXERCISE_BASIC_OBJECT_METHODS already calls to an objects PrintSelf
method.

Refactor the tests for the sake of coherency across ITK: remove white
spaces in excess, improve indentation.

Also, remove the itkStrainImageFilterEulerianTest.cxx and
itkStrainImageFilterLagrangianTest.cxx files: all supported strain forms
are now tested in itkSTrainImageFilterTest.cxx.

### CODE REVIEW

**Comment by jhlegarreta** [comment]
@thewtex Tested locally and all tests pass. Further reviews of the patch are welcome.

File: test/itkTransformToStrainFilterTest.cxx [hunk 1]
```diff
@@ -22,33 +22,70 @@
...
+  transformToStrainFilter->SetStrainForm( strainForm );
+
+  TRY_EXPECT_EXCEPTION( transformToStrainFilter->Update() );
+
+
+  // Get the input strain form
+  if( !strcmp( argv[1], "INFINITESIMAL" ) )
     {
-    strainForm = 0;
+    strainForm = TransformToStrainFilterType::StrainFormType::INFINITESIMAL;
```
Comment (thewtex) [inline test/itkTransformToStrainFilterTest.cxx]: Could these please still be kept as `0`, `1`, `2` -- the compiler can complain or error if C++11 is not enabled.

**Comment by jhlegarreta** [comment]
@thewtex Thanks Matt. Try the new commit.

**Comment by thewtex** [comment]
Jon, it looks like the improved coverage uncovered a few issues :-).  Addressed in some follow-up commits here:

  https://github.com/KitwareMedical/ITKStrain/pull/8

**Pull request closed without merging**